docs/design/satellite-svc-separation.md

```diff
@@ -0,0 +1,208 @@
+# Satellite Service Separation
+
+## Table of Contents
+1. [Abstract](#abstract)
+2. [Background](#background): Current Satellite Design
+3. [Design](#design)
+4. [Rationale](#rationale)
+4. [Implementation](#implementation)
+5. [Open Issues](#open-issues)
+
+## Abstract
+
+The goal of the design doc is to describe the work necessary to make the Satellite horizontally scalable.
+
+## Background
+
+Currently all Satellite services run in a single binary in a single process. While this is great for development, this is bad for scaling.
```

Comment:
Throughout the document only use "process" for clarity. It will be still one binary, probably.